sources/api/prairiedog/src/main.rs

```diff
@@ -253,8 +286,119 @@ fn load_crash_kernel() -> Result<()> {
     Ok(())
 }
 
+/// Queries Bottlerocket boot settings and generates initrd image file with boot config as the only data
+async fn generate_boot_config<P>(socket_path: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let boot_settings = get_boot_config_settings(socket_path).await?;
+    info!("Generating initrd boot config from boot settings");
+    trace!("Boot settings: {:?}", boot_settings);
+    let bootconfig = to_string(&boot_settings).context(error::SerializeBootConfigSnafu)?;
+    trace!("Serializied boot config string: {}", bootconfig);
+    let initrd = generate_initrd(&bootconfig.into_bytes())?;
+    trace!("Writing initrd image file: {:?}", initrd);
+    tokio::fs::write(BOOTCONFIG_INITRD_PATH, &initrd)
+        .await
+        .context(error::WriteInitrdSnafu)?;
+    Ok(())
+}
+
+async fn get_boot_config_settings<P>(socket_path: P) -> Result<BootSettings>
+where
+    P: AsRef<Path>,
+{
+    let boot_settings = schnauzer::get_settings(socket_path)
```

Comment:
Is `schnauzer` the default client to get settings from the API? We don't use it across the binaries that interact with the API. Should we standardize it as the client going forward?

Reply:
We're just borrowing `schnauzer`'s interface for getting API settings which itself is a thin wrapper around `apiclient::raw_request`. We don't use it across all the binaries since `schnauzer` was introduced later than the `apiclient` library.

I don't think we need to enforce a "standard", there are other use cases where it's helpful to directly use the `apiclient` library call. In this case though, `schnauzer::get_settings` fits our use-case so we use it.